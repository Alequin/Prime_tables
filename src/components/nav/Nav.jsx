import React from "react";

class Nav extends React.Component{

  constructor(props){
    super(props)
  }

  render(){
    return (
      <div>
        Nav Bar
      </div>
    )
  }
}

export default Nav
