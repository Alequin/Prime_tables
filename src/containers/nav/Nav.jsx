import React from "react"
import PropTypes from 'prop-types'

class Nav extends React.Component{

  constructor(props){
    super(props)
    this.handleButtonClick = this.handleButtonClick.bind(this)
    this.onChangeInputBox = this.onChangeInputBox.bind(this)
    this.state = {
      inputText: ""
    }
  }

  componentDidMount(){
    this.inputBox = document.querySelector(".input")
  }

  handleButtonClick(event){
    event.preventDefault()
    if( !(/\./.test(this.state.inputText)) ){
      this.inputBox.style.borderColor = ""
      this.props.onClickButton(this.state.inputText)
    }else{
      this.inputBox.style.borderColor = "red"
    }
  }

  onChangeInputBox(event){
    this.setState({
      inputText: event.target.value
    })
  }

  render(){
    return (
      <nav className="nav-bar">
        <form>
          <label>
            <h2>Enter how many prime numbers you wish to generate</h2>
            <div className="align-center-container">
              <input
                className="input"
                type="number"
                step=".01"
                value={this.state.inputText}
                onChange={this.onChangeInputBox}
              />
            </div>
          </label>
          <div className="align-center-container">
            <input
              className="button"
              type="submit"
              value="Generate prime number multiplication table"
              onClick={this.handleButtonClick}
            />
          </div>
        </form>
      </nav>
    )
  }
}

Nav.propTypes = {
  onClickButton: PropTypes.func,
}

export default Nav
