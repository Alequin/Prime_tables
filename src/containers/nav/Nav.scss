.nav-bar{
  position: fixed;
  background-color: #228B22;
  width: 100%;

  h2{
    color: white;
    text-align: center;
  }
}
