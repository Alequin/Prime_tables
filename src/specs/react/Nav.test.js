import React from 'react';
import Nav from './../../components/nav/Nav.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Nav/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
