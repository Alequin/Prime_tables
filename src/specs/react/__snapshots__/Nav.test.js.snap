// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="nav-bar"
>
  <label>
    <h2>
      Enter how many prime numbers you wish to generate
    </h2>
    <div
      className="align-center-container"
    >
      <input
        className="input"
        onChange={[Function]}
        step=".01"
        type="number"
        value="3"
      />
    </div>
  </label>
  <div
    className="align-center-container"
  >
    <input
      className="button"
      onClick={[Function]}
      type="submit"
      value="Generate prime number multiplication table"
    />
  </div>
</nav>
`;
